src/compas/robots/resources/basic.py

```diff
@@ -55,12 +55,34 @@ def load_mesh(self, url):
 
 
 class DefaultMeshLoader(AbstractMeshLoader):
-    """Handles basic mesh loader tasks, mostly from local files."""
+    """Handles basic mesh loader tasks, mostly from local files.
 
-    def __init__(self):
+    Attributes
+    ----------
+    kwargs (optional): dict
+        Additional keyword arguments.
+    """
+
+    def __init__(self, **kwargs):
         super(DefaultMeshLoader, self).__init__()
+        self.attr = dict() or kwargs
 
     def can_load_mesh(self, url):
+        """Determine whether this loader can load a given mesh URL.
+
+        Parameters
+        ----------
+        url : str
+            Mesh URL.
+
+        Returns
+        -------
+        bool
+            ``True`` if the URL points to a local and valid file.
+            Otherwise ``False``.
+        """
+
+        url = self._get_local_path(url) or url
```

Comment:
Minor detail, but I would make `_get_local_path()` return `url` instead of `None`, so that client code is simpler instead of having to know how to select the default.

Reply:
Agreed.  Now `_get_local_path()` would look like:
```
    def _get_local_path(self, url):
        basepath = self.attr.get('basepath') 
        if basepath:
            return os.path.join(basepath, url)
        return url
```

Reply:
👍